src/components/LoginModal/index.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { FC, useState } from 'react';
+import { Modal } from 'react-overlays';
+import { Text } from '@shared/Text';
+import { Button } from '@shared/Button';
+import { Checkbox } from '@shared/Checkbox';
+import { ButtonCancel } from '@shared/ButtonCancel';
+import { SubmitHandler, useForm } from 'react-hook-form';
+import {
+  LoginInputsType,
+  LoginModalPropsType,
+} from '@/components/LoginModal/type';
+import { getDataFromLS } from '@/helpers/getStartUsers';
+import { IUser } from '@/context/userContext';
+
+export const LoginModal: FC<LoginModalPropsType> = ({
+  isOpen,
+  renderBackdrop,
+  close,
+  registerOpen,
+}) => {
+  const [isRemembered, setIsRemembered] = useState(false);
+  const [usersLS] = useState(getDataFromLS('rememberedUsers'));
+  const credential = getDataFromLS('credential');
+  const addUsers = getDataFromLS('theAllUsers');
+
+  const {
+    register,
+    handleSubmit,
+    reset,
+    formState: { errors },
+  } = useForm<LoginInputsType>({
+    defaultValues: {
+      username: '',
+      password: '',
+    },
+  });
+  const onSubmit: SubmitHandler<LoginInputsType> = ({ username, password }) => {
+    if (isRemembered) {
+      const withNewUser = usersLS.concat(username);
+      localStorage.setItem('rememberedUsers', JSON.stringify(withNewUser));
+      if (
+        (credential.username === username &&
+          credential.password === password) ||
+        addUsers.find((person: IUser) => person.username === username)
+      ) {
+        localStorage.setItem('currentUser', JSON.stringify(username));
+        reset();
+        setIsRemembered(false);
+        close();
+      }
+    }
+
+    if (
```

Comment:
There's too much logic in this component, I'd want you to move everything in useTwitter() hook.